### ISSUE TITLE
`NonZero` parameter types

### ISSUE THREAD

**Issue by grarco** [anoma/namada#3347]
We should consider changing some of the primitive integer parameters to their `NonZero` equivalent 

_Originally posted by @sug0 in https://github.com/anoma/namada/pull/3318#discussion_r1622446820_

**Comment by brentstone** [comment]
how meticulous should we be about this? If we want, we can change a lot of protocol and genesis params?

**Comment by cwgoes** [comment]
Does this type change also change the formatting which must be used in the genesis parameter files?

**Comment by tzemanovic** [comment]
I don't think it changes format- they have the same layout and bit validity except for 0.

That being said, I don't know if using `NonZero` types gives us much benefit except for rejecting 0. Alternatively we could put any invariants in `fn validate_parameters` in `crates/apps_lib/src/config/genesis/templates.rs` and each VP that has parameters should also guard its invariants in its VP implementation (see e.g. `crates/proof_of_stake/src/parameters.rs` `fn validate`). I think this is more flexible as we can express more rules than just non-zero.

**Comment by Fraccaman** [comment]
Okay makes sense I think we can close this one then.
